src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -362,6 +397,38 @@ class JSONReader
         }
     }
 
+    bool IsRemote()
+    {
+        return !_remoteUrl.empty();
+    }
+
+    std::string ResolveRemoteUri(const std::string &uri){
+        std::string scheme, path, fragment;
+        ParseURI(uri, scheme, path, fragment );
+
+
+        if (scheme == "file") {
+            return ResolveURI(scheme, path, std::string(), _vOpenRAVESchemeAliases);
+        }
+        else if (_StartsWith(uri, "http")) {
+            return uri;
+        }
+        else if (IsRemote()) {
+            return _remoteUrl + uri.substr(uri.find_first_of(":/")+2,  uri.find_last_of("#") - uri.find_last_of(":/") -1);
```

Comment:
I do not agree with this design. You should not use `_remoteUrl` until right before calling `curl`.

`_StartsWith(uri, "http")` is a very dangerous check, you are preventing loading of a scene like `http.mujin.msgpack`.

In the future, there could be other protocols, like `remoteUrl = ftp://....` or even `remoteUrl = files://...`, you cannot assume `http`

Reply:
removed starts with, need to implement other protocols such as ftp://, 
file:// currently works